kong/plugins/rate-limiting/schema.lua

```diff
@@ -23,6 +23,52 @@ local function validate_periods_order(config)
 end
 
 
+local function is_dbless()
+  local ok, database = pcall(function()
+    return kong.configuration.database
+  end)
+
+  if not ok then
+    return nil
+  end
+
+  local ok, role = pcall(function()
+    return kong.configuration.role
+  end)
+
+  if not ok then
+    return nil
+  end
+
+  return database == "off" or role == "control_plane" or role == "data_plane" or nil
+end
```

Comment:
Just curious, any chance we can make this a PDK function? It seems to be a lot of code we are copying in this case...

Reply:
I just made the function a bit smaller, and it is much less bothersome to copy-paste it in two places now. Maybe if we need it in more places we can abstract it.